### ISSUE TITLE
Unable to add ZFS pool.

### ISSUE THREAD

**Issue by slavachertanovo** [SaumonNet/proxmox-nixos#28]
Datacenter => Storage => Add => ZFS
No pools visible there.

**Comment by ChanningHe** [comment]
Same problem. 
```
command 'zfs list -t filesystem -Hp -o name,avail.used' failedopen3:exec of zfs list -t filesystem -Hp -o name.avail.usedfailed: No such file or directory at/nix/store/m9jlc7smvy7wi2zvnb1pbn4hh61h1909-per1-5.36.3-
```
And if you edit /etc/pve/storage.cfg to add zfs in cli, I will encounter the following error.
```
could not activate storage 'local-zfs', zfs error: open3: exec of zpool import -d /dev/disk/by-id/ -o cachefile=none nixpool failed: No such file or directory at /nix/store/m9jlc7smvy7wi2zvnb1pbn4hh61hl909-perl-5.36.3-env/lib/perl5/site_perl/5.36.3/PVE/Tools.pm line 494. (500)
```

**Comment by codgician** [comment]
The reason for this is command arguments of `open3` calls in the original perl scripts are not patched.

For example: https://github.com/proxmox/pve-storage/blob/ec5395868f5f24c0dcddd3f7e572db337160706e/src/PVE/Storage.pm#L1415

This could be fixed by either patching the argument of these `open3` patterns when building `pve-storage`, or adding all possible binaries to PATH in corresponding systemd services.

**Comment by codgician** [comment]
I created a quick fix with PR #72 to workaround this issue, which is equivalent to:

```nix
systemd.services.pvedaemon.path = [ pkgs.zfs ];
systemd.services.pvestatd.path = [ pkgs.zfs ];
```

Feel free to review and comment your thoughts.
